psi4/driver/procrouting/empirical_dispersion.py

```diff
@@ -87,22 +85,23 @@ def __init__(self, alias, dtype, **kwargs):
                 if len(tuple_params) > 4:
                     raise Exception("Too many parameter in input tuple param.")
 
-        else: # Only other case at the moment
+        elif self.dtype in ["-NL"]:
+            self.disp_type = 'nl'
+
+        else:  # Only other case at the moment
             self.dtype = self.dtype.replace('-D2P4', '-D2')
             self.disp_type = 'p4'
             if tuple_params is not None:
                 self.dash_params = {}
                 for k, v in zip(['s6', 'p1', 'p2', 'p3'], tuple_params):
                     self.dash_params[k] = v
 
-
         # Build the C++ dispersion class
         if self.disp_type == 'p4':
             self.disp = core.Dispersion.build(self.dtype, **self.dash_params)
         else:
             self.disp = None
```

Comment:
This probably needs expanded so that the `compute_things()` functions below don't have to return 0.